### PR TITLE
changed over to id

### PR DESCRIPTION
**Pull request by Searay-330**
#203 please review this commit very carefully it makes some sweeping changes

### CODE REVIEW

**Review by adhishchugh — COMMENTED** [review]
Wait, why are you using _id and not just id?
Can you not see the id field?

**Comment by Searay-330** [comment]
because its _id in the db so thats what i get in the request object

**Comment by adhishchugh** [comment]
You should also get id. _id is usually used for internal backend use only.

**Comment by Searay-330** [comment]
what i get is _id

**Comment by theknickerbocker** [comment]
So if you look at the route to the submission page it's something like:

'http://localhost:8000/courses/testcourse-f17/assignments/null'

So I'm not sure what value is used for the route (I think it was assignment_num), but now it is null which messes up the routes. 

Then I'm getting this error as well
screen shot 2017-11-03 at 12 57 08 pm
since multiple assignments have the same value for their route.

**Comment by Searay-330** [comment]
i didnt notice this i will look more into it ill keep you posted.

**Comment by theknickerbocker** [comment]
Sounds good, was looking into it and it looks like the id value is null on all assignments

**Comment by theknickerbocker** [comment]
Submissions don't show up in the uploads of BE

**Comment by theknickerbocker** [comment]
Still getting same bug

**Comment by theknickerbocker** [comment]
Submissions show up now but the directory creation on backend isn't right:

screen shot 2017-11-04 at 1 05 43 pm

Which then results in the server crashing:

screen shot 2017-11-04 at 1 07 29 pm

File: client/modules/Gradebook/Gradebook.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ export class Gradebook extends Component {
 
                     {data.map((n, index) => (
                         <TableRow key={index} >
-                            <TableRowColumn>{this.props.assignmentsMapping[n.assignment_num].name}</TableRowColumn>
+                            <TableRowColumn>{this.props.assignmentsMapping[n.id].name}</TableRowColumn>
```
Comment (theknickerbocker) [inline client/modules/Gradebook/Gradebook.js]: change this back

**Comment by theknickerbocker** [comment]
i made a comment on a line that should be changed back to what it previously was

**Review by theknickerbocker — APPROVED** [review]
Everything works, nice job

**Pull request merged**